### ISSUE TITLE
0.8 from crates.io fails to build

### ISSUE THREAD

**Issue by gmorenz** [sebcrozet/ncollide#131]
Trying to build 0.8 from crates.io fails with the following error message, this breaks nphysics. The 0.8.0 release on github builds fine, and changing to using that makes nphysics compile as well.

Cargo.toml:
```
[package]
name = "tmp2"
version = "0.1.0"
authors = ["morenzg <morenzg@gmail.com>"]

[dependencies]
ncollide = "0.8.*"
```

Error message:
```
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/compound.rs:36:20: 36:40 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/compound.rs:36           M:       Isometry<P, P::Vect> {
                                                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/compound.rs:36:20: 36:40 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:64:14: 64:34 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:64           M: Isometry<P, P::Vect> {
                                                                                                                                                           ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:64:14: 64:34 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:97:14: 97:34 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:97           M: Isometry<P, P::Vect> {
                                                                                                                                                           ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/shape/composite_shape.rs:97:14: 97:34 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/aabb_repr.rs:11:14: 11:34 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/aabb_repr.rs:11           M: Isometry<P, P::Vect> {
                                                                                                                                                               ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/aabb_repr.rs:11:14: 11:34 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_compound.rs:10:15: 10:35 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_compound.rs:10           M:  Isometry<P, P::Vect>,
                                                                                                                                                                               ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_compound.rs:10:15: 10:35 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_repr.rs:12:14: 12:34 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_repr.rs:12           M: Isometry<P, P::Vect> {
                                                                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/bounding_volume/bounding_sphere_repr.rs:12:14: 12:34 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36                   M: Isometry<P, P::Vect> {
                                                                                                                                                                                 ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 note: in this expansion of impl_composite_shape_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36                   M: Isometry<P, P::Vect> {
                                                                                                                                                                                 ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 note: in this expansion of impl_composite_shape_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36                   M: Isometry<P, P::Vect> {
                                                                                                                                                                                 ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 note: in this expansion of impl_composite_shape_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_composite_shape.rs:36:22: 36:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 error: wrong number of type arguments: expected 1, found 2 [E0244]
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37                   M: Isometry<P, P::Vect> {
                                                                                                                                                                             ^~~~~~~~~~~~~~~~~~~~
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 note: in this expansion of impl_support_map_repr! (defined in /home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs)
/home/normal/.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/ncollide_entities-0.5.0/inspection/maybe_as_support_map.rs:37:22: 37:42 help: run `rustc --explain E0244` to see a detailed explanation
error: aborting due to 17 previous errors
error: Could not compile `ncollide_entities`.

To learn more, run the command again with --verbose.
```

**Comment by sebcrozet** [comment]
Jeez, I should have changed the minor version number of `ncollide_math` instead of the path version number… I'll fix that and update nphysics at the same time.

**Comment by sebcrozet** [comment]
I've yanked all the versions that will fail to compile because of this mistake. So ncollide 0.8.0 and nphysics 0.3.0 from crates.io should compile again after a `cargo update`. Sorry for that!

I've published nphysics 0.4.0 which now uses ncollide 0.9.0.
